src/main/java/io/fabric8/maven/docker/util/DockerFileUtil.java

```diff
@@ -166,10 +166,14 @@ public static FixedStringSearchInterpolator createInterpolator(MojoParameters pa
     }
 
     static String resolveArgValueFromStrContainingArgKey(String argString, Map<String, String> args) {
-        if (argString.startsWith("$") && args.containsKey(argString.substring(1))) {
-            return args.get(argString.substring(1));
-        } else if (argString.startsWith("${") && argString.endsWith("}") && args.containsKey(argString.substring(2, argString.length() - 1))) {
-            return args.get(argString.substring(2, argString.length() - 1));
+        Pattern argPattern = Pattern.compile("\\$(?:\\{(.*)\\}|(.*))");
```

Comment:
```suggestion
        Pattern argPattern = Pattern.compile("^\\$(?:\\{(.*)\\}|(.*))$");
```

Align with start and end of the string. Otherwise you would also match `bla$ad` or `${foo}bar`

Reply:
But please verify my assumption within the tests, too, please